### DISCUSSION TITLE
Creality Halot One automatic upload

### DISCUSSION THREAD

**Discussion by xPew** [sn4k3/UVtools#394] [Q&A]
I just bought a Creality Halot One, and it has WiFi capabilities. By default it supports Creality cloud. I am not sure how close this is, but an option to upload from UVtools to the printer directly would be awesome. Any chance doing this? Or it would need firmware modification at Creality's side?
It's connected to the local network independent from creality cloud app, what can be used to update the firmware and browse some models, than there is an additional option to connect it to the creality cloud app, but that's their proprietary app for browsing, slicing and uploading, which is nice but I love PS and UVtools more :)

**Comment by tslater2006** [comment]
Pretty sure this was added already to uvtools.

**Comment by tslater2006** [comment]
Looks like it was added in v2.22.0. https://github.com/sn4k3/UVtools/commit/459e7ca0faacd1cd98cad9d49fd918fd3390b387

Unless that was for some other networked printers and the Halot needs something different.

**Reply by xPew** [reply]
I get this msg: No connection could be made because the target machine actively refused it. (IP)
This case its a kind of close wifi?

**Comment by sn4k3** [comment]
I don't have the details of the comunication or how it can be done. Unless someone share how files can be send it will not be possible to add that to UVtools. Only Bene4 is compatible with that function atm
